source/agora/consensus/EnrollmentPool.d

```diff
@@ -84,35 +87,42 @@ public class EnrollmentPool
             return false;
         }
 
-        // check if already exists
-        if (this.hasEnrollment(enroll.utxo_key))
-        {
-            log.info("Rejected already existing enrollment, Data was: {}",
-                enroll);
-            return false;
-        }
-
-        static ubyte[] buffer;
-        try
-        {
-            serializeToBuffer(enroll, buffer);
-        }
-        catch (Exception ex)
-        {
-            log.error("Serialization error: {}, Data was: {}", ex.msg, enroll);
-            return false;
-        }
-
         try
         {
-            () @trusted {
-                this.db.execute("INSERT INTO enrollment_pool (key, val) VALUES (?, ?)",
-                    enroll.utxo_key.toString(), buffer);
-            }();
+            if (this.hasEnrollment(enroll.utxo_key))
+            {
+                () @trusted {
+                    this.db.execute("UPDATE enrollment_pool SET " ~
+                        "avail_height = ? WHERE key = ?",
+                        avail_height.value, enroll.utxo_key.toString());
+                }();
+                return false;
+            }
+            else
+            {
+                static ubyte[] buffer;
+                try
+                {
+                    serializeToBuffer(enroll, buffer);
+                }
+                catch (Exception ex)
+                {
+                    log.error("Serialization error: {}, Data was: {}", ex.msg,
+                        enroll);
+                    return false;
+                }
+                () @trusted {
+                    this.db.execute("INSERT INTO enrollment_pool " ~
+                        "(key, val, avail_height) VALUES (?, ?, ?)",
+                        enroll.utxo_key.toString(), buffer,
+                        avail_height.value);
+                }();
+            }
```

Comment:
https://www.sqlitetutorial.net/sqlite-replace-statement/

Reply:
I will apply it to the code. Thanks.